### DISCUSSION TITLE
How to I type the onChange function for the Select

### DISCUSSION THREAD

**Discussion by AmodeusR** [JedWatson/react-select#5363] [Q&A] [answered]
This is really confusing and the page about typescript doesn't make anything clear. I'm trying to get the value of the select through a useState, but passing the set function throws a type error. In the website it doesn't tell us how to properly type the function, which it take those "Option", "actionMeta" out of nowhere inside a function declaration but no useState or anything similar.

Can anyone tell me how am I suppose to type onChange function in an easy / well explained way?

**Comment by Methuselah96** [comment, marked as answer]
Here's a [simple example](https://codesandbox.io/s/nice-lena-wmb7o1?file=/src/App.tsx). Let me know if you're wanting to type a more complex or different case.

**Reply by AmodeusR** [reply]
Perfectly what I needed! Thank you very much for your help and patience. This example should be put on the page about Typescript typing...
